src/main/jbake/content/blog/2017/12/running-agent.adoc

```diff
@@ -0,0 +1,155 @@
+...
+JAVA_OPTS="$JAVA_OPTS -javaagent:$JBOSS_HOME/bin/hawkular-javaagent.jar=config=$JBOSS_HOME/standalone/configuration/hawkular-javaagent-config.yaml,delay=10"
+
+```
+
+If you are running WildFly in domain mode, the domain.conf you should have something like:
+
+```
+HOST_CONTROLLER_JAVA_OPTS="$HOST_CONTROLLER_JAVA_OPTS -Dhawkular.agent.metadata.domain=true \
+   -Djboss.modules.system.pkgs=org.jboss.byteman,org.jboss.logmanager \
+   -Djava.util.logging.manager=org.jboss.logmanager.LogManager \
+   -Djboss.host.server-excluded-properties=jboss.modules.system.pkgs,java.util.logging.manager,hawkular.agent.metadata.domain \
+   -javaagent:$JBOSS_HOME/bin/hawkular-javaagent.jar=config=$JBOSS_HOME/domain/configuration/hawkular-javaagent-config-domain.yaml,delay=10"
+```
+
+==== Running in WildFly Domain Mode
+
+When running WildFly in domain mode, a Hawkular Agent runs in the host controller.
+There also needs to be a Hawkular Agent running in each of the slave servers spawned by the host controller. You ensure
+an agent is attached to each slave server by adding the proper -javaagent command line option to the
+server JVM command found in the host.xml. Note also that you also have to tell each slave server's agent what port it
+should bind the metrics exporter endpoint to. For example:
+
+```
+  <jvms>
+    <jvm name="default">
+      <jvm-options>
+        ...
+        <!-- HAWKULAR SETTINGS -->
+        <option value="-Djboss.modules.system.pkgs=org.jboss.byteman,org.jboss.logmanager"/>
+        <option value="-Djava.util.logging.manager=org.jboss.logmanager.LogManager"/>
+        <option value="-javaagent:${jboss.home.dir}/bin/hawkular-javaagent.jar=config=${jboss.domain.config.dir}/hawkular-javaagent-config-metrics-only.yaml,delay=10"/>
+      </jvm-options>
+    </jvm>
+  </jvms>
+  <servers>
+    <server name="server-one" group="main-server-group">
+      <jvm name="default">
+        <jvm-options>
+          <!-- HAWKULAR SETTINGS -->
+          <option value="-Dhawkular.agent.metrics.port=9780"/>
+        </jvm-options>
+      </jvm>
+    </server>
+  </servers>
+  ...
+```
+
+The slave server agents are to be run in metrics-only mode (that is, no managed-servers section is defined in their configuration files).
+
+=== Example
+
+To see an example WildFly distribution configured to run the Hawkular Agent in either standalone or domain mode,
+build from source or grab from a Maven repository the Maven artifact "org.hawkular.agent:hawkular-javaagent-wildfly-dist".
+Look at its `bin/` directory for the domain.conf and standalone.conf files and look in the
+`standalone/configuration` and `domain/configuration` directories for the different agent configuration files. Note also
+the `host-hawkular.xml` that is to be used with the `--host-config` option when running in domain mode.
+
+== How It Works
+
+When the agent starts, it reads its configuration file from disk. The agent then attempts to connect to the Hawkular Server.
```

Comment:
"When the agent starts"

We've been talking about multiple agents in different places and I think it would be helpful to be very specific which one we're talking about so that people don't get confused.  Are we talking about the agent on the host controller here, or one of the agents on a managed/slave server?

Reply:
In this specific case, its talking about ANY agent (hence, no qualifier here) - no matter if the agent is in standalone server, host controller, slave server, or any non-wildfly JVM, this paragraph applies.